Octokit/Clients/IIssuesClient.cs

```diff
@@ -163,5 +163,25 @@ public interface IIssuesClient
         /// </param>
         /// <returns></returns>
         Task<Issue> Update(string owner, string name, int number, IssueUpdate issueUpdate);
+
+        /// <summary>
+        /// Locks an issue for the specified repository. Issue owners and users with push access can lock an issue.
+        /// </summary>
+        /// <remarks>https://developer.github.com/v3/issues/#lock-an-issue</remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="number">The issue number</param>
+        /// <returns></returns>
+        Task<Issue> Lock(string owner, string name, int number);
```

Comment:
Given that the API does return a HTTP 204 NoContent response when "succesful" the way ive seen this implemented in octokit in other places (eg [OrganizationMembers.Publicize()](https://github.com/octokit/octokit.net/blob/829677c9292ca503806aac385f49e37ad3730fdd/Octokit/Clients/OrganizationMembersClient.cs#L289)) is that the Octokit call returns a Task<bool> then internally it calls the `Connection.Put` and checks the HTTP response.  If a 204 was received then "true" is returned, otherwise false (or an exception of course, for 404/500/other stuff handled by the plumbing inside Connection class)